genesis/public/core/instance.go

```diff
@@ -0,0 +1,161 @@
+/*
+ *    Copyright 2018 INS Ecosystem
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package core
+
+import (
+	"fmt"
+
+	"github.com/insolar/insolar/genesis/model/class"
+	"github.com/insolar/insolar/genesis/model/domain"
+	"github.com/insolar/insolar/genesis/model/factory"
+	"github.com/insolar/insolar/genesis/model/object"
+)
+
+// InstanceDomainName is a name for instance domain.
+const InstanceDomainName = "InstanceDomain"
+
+// InstanceDomain is a contract that stores instances of other domains
+type InstanceDomain interface {
+	// Base domain implementation.
+	domain.Domain
+	// CreateInstance is used to new instance as a child to domain storage.
```

Comment:
You probably forgot word "create" here :) ("is used to **create** new instance")